test/distributed/_shard/test_sharder.py

```diff
@@ -0,0 +1,151 @@
+        for i in range(ebc.num_bags):
+            bag_key = f"embedding_bag_{i}"
+            if i < self.split_idx:
+                shard_module(ebc, plan=ShardingPlan(plan={f"embedding_bags.{bag_key}.weight": row_spec}))
+            else:
+                shard_module(ebc, plan=ShardingPlan(plan={f"embedding_bags.{bag_key}.weight": col_spec}))
+
+            self.embedding_bags[bag_key] = ebc.embedding_bags[bag_key]
+
+
+class CustomSharder(Sharder):
+    def __init__(self, devices, split_sharding_idx):
+        self.devices = devices
+        self.split_sharding_idx = split_sharding_idx
+        self.rowwise_spec = ChunkShardingSpec(dim=0, placements=devices)
+        self.colwise_spec = ChunkShardingSpec(dim=1, placements=devices)
+
+    def shard(self, ebc: nn.Module) -> nn.Module:
+        if not isinstance(ebc, CustomEmbeddingBagCollection):
+            raise RuntimeError("The custom sharder only supports CustomEmbeddingBagCollection")
+
+        return CustomShardedEBC(ebc, self.split_sharding_idx, (self.rowwise_spec, self.colwise_spec))
+
+
+class TestCustomSharder(ShardedTensorTestBase):
+
+    @with_comms(init_rpc=False)
+    @skip_if_lt_x_gpu(TEST_GPU_NUM)
+    @requires_nccl()
+    def test_custom_sharder(self):
+        class MyModule(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.ebc = CustomEmbeddingBagCollection(10, 10, 8)
+
+            def forward(self, inputs):
+                return self.ebc(inputs)
+
+        custom_sharder = CustomSharder(
+            devices=[f"rank:{i}/cuda:{i}" for i in range(TEST_GPU_NUM)],
+            split_sharding_idx=TEST_GPU_NUM // 2
+        )
+
+        sharding_plan = ShardingPlan(
+            plan={
+                "ebc": custom_sharder,
+            })
+
+        local_model = MyModule().cuda(self.rank)
+        sharded_model = copy.deepcopy(local_model)
+
+        # shard the module with the provided sharding plan
+        shard_module(sharded_model, sharding_plan)
+
+        # check to make sure the module already been sharded
+        emb_bags = sharded_model.ebc.embedding_bags
+        self.assertTrue(isinstance(emb_bags["embedding_bag_0"].weight, ShardedTensor))
+        self.assertTrue(isinstance(emb_bags["embedding_bag_9"].weight, ShardedTensor))
+        self.assertEqual(emb_bags["embedding_bag_0"].weight.sharding_spec(), custom_sharder.rowwise_spec)
+        self.assertEqual(emb_bags["embedding_bag_9"].weight.sharding_spec(), custom_sharder.colwise_spec)
+
+        # make sure we can run sharded computation and compare outputs
+        # with the local model version
+        input = torch.arange(8).reshape((2, 4)).cuda(self.rank)
+        local_output = local_model(input)
+        sharded_output = sharded_model(input)
+
+        self.assertEqual(local_output, sharded_output)
+
+    @with_comms(init_rpc=False)
+    @skip_if_lt_x_gpu(TEST_GPU_NUM)
+    @requires_nccl()
+    def test_custom_sharder_errors(self):
```

Comment:
Can we also test the case where a custom sharder and an entry in the sharding plan conflict? We should error out in such cases.